docs/examples/io_tests/test_cases/cpp/setup/compile.sh

```diff
@@ -13,19 +13,19 @@
 unset compilationStatus
 
 #language specific compile and run of each test case
-g++ -std=c++11 *.cpp 2>&1 | tee $testLog > /dev/null
-compilationStatus=${PIPESTATUS[0]}
+g++-6 -std=c++11 ./*.cpp 2>&1 | tee "$testLog" > /dev/null
```

Comment:
you can put a shellcheck ignore before this line. The ignore statement is:
\# shellcheck disable=SC2154

you can also modify the statement to:
g++-6 -std=c++11 "./*.cpp" 2>&1 | tee "$testLog" > /dev/null